testssl.sh

```diff
@@ -17194,7 +17194,12 @@ EOF
 }
 
 maketempf() {
-     TEMPDIR=$(mktemp -d /tmp/testssl.XXXXXX) || exit $ERR_FCREATE
+     TEMPDIR=$(mktemp -d /tmp/testssl.XXXXXX)
+     if [[ $? -ne 0 ]]; then
+          # for e.g. devices where we can't write to /tmp:
+          TEMPPATH=$PWD
+          TEMPDIR=$(mktemp -d $PWD/testssl.XXXXXX) || exit $ERR_FCREATE
```

Comment:
Hi Dirk,

I think the `TEMPPATH=$PWD` line can just be removed. As far as I can tell, `TEMPPATH` is not currently defined in testssl.sh, and this PR doesn't make use of it, so there doesn't seem to be a need to define it.

As someone noted in #1277, there is a possibility that the current directory ($PWD) contains spaces in it, which could cause problems since at the moment $TEMPDIR is used in many places without quoting it. We could go through and add quotes wherever `$TEMPDIR` (or any variable such as TMPFILE, ERRFILE, HEADERFILE, SOCK_REPLY_FILE, HOSTCERT, etc.) is used. Barring that, however, it might be a good idea to check whether `$PWD` contains any white space characters, and then issue an error if /tmp is not writable and `$PWD` contains white space.

Reply:
Still thinking on it not sure whether this is enough

``mkdir "ls;" ; cd "ls;" ``. What else should we not allow? Maybe an alphanumerical white list is just better plus "/"